unity-renderer/Assets/DCLFeatures/CameraReel/Gallery/Scripts/CameraReelGalleryView.cs

```diff
@@ -0,0 +1,119 @@
+﻿using Cysharp.Threading.Tasks;
+using DCLServices.CameraReelService;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using UnityEngine;
+using UnityEngine.Networking;
+using UnityEngine.UI;
+
+namespace DCLFeatures.CameraReel.Gallery
+{
+    public class CameraReelGalleryView : MonoBehaviour
+    {
+        private readonly Dictionary<int, GridContainerComponentView> monthContainers = new ();
+        private readonly Dictionary<CameraReelResponse, GameObject> screenshotThumbnails = new ();
+
+        [SerializeField] private RectTransform container;
+        [SerializeField] private Button showMoreButton;
+        [SerializeField] private RectTransform showMoreButtonPanel;
+        [SerializeField] private Canvas canvas;
+
+        [Header("RESOURCES")]
+        [SerializeField] private GameObject monthHeaderPrefab;
+        [SerializeField] private GridContainerComponentView monthGridContainerPrefab;
+        [SerializeField] private Image thumbnailPrefab;
+
+        private GridContainerComponentView currentMonthGridContainer;
+
+        public event Action<CameraReelResponse> ScreenshotThumbnailClicked;
+        public event Action ShowMoreButtonClicked;
+
+        private void Awake()
+        {
+            SwitchVisibility(isVisible: false);
+        }
+
+        private void OnEnable()
+        {
+            showMoreButton.onClick.AddListener(() => ShowMoreButtonClicked?.Invoke());
```

Comment:
We could just do it in the awake instead of adding and removing every enable/disable, right? Or am I missing something